pkg/core/index.go

```diff
@@ -11,50 +9,11 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
-// indexIterator iterates over a list of index files,
-// that is files containing BundleEntries referring to
-// hashes on the blob storage.
-type indexIterator interface {
-	Next() string
-}
-
-// defaultIndexIterator implements indexIterator with a known max number of index files
-// and a path function normally provided by the metadata model package
-// (e.g. model.GetArchivePathToBundleFileList).
-type defaultIndexIterator struct {
-	i, max    uint64
-	exclusive sync.Mutex
-	pather    func(uint64) string
-}
-
-func (di *defaultIndexIterator) Next() string {
-	di.exclusive.Lock()
-	defer di.exclusive.Unlock()
-	if di.i >= di.max {
-		return ""
-	}
-	di.i++
-	return di.pather(di.i - 1)
-}
-
-func newDefaultIndexIterator(max uint64, pather func(uint64) string) *defaultIndexIterator {
-	return &defaultIndexIterator{
-		max:    max,
-		pather: pather,
-	}
-}
-
-type patherIterator interface {
-	// Next yields the next series of index files to retrieve, with some reusable identifier
-	// and an iterator to walk through individual numbered index files.
-	//
-	// When completed, it returns ("", 0,nil).
-	//
-	// This iterator may be called concurrently.
-	Next() (string, indexIterator)
-}
-
-// fileIndex builds an index of files from file lists in metadata.
+// fileIndex manages an index of files based on file lists stored in metadata.
+//
+// Indexing is critical to be able to retrieve files from the blob storage.
+//
+// File lists are split into manageable chunks. By default each single index file holds up to 1000 file entries.
```

Comment:
based on this comment, i now understand that "index" throughout this patch is a synonym (or near-synonym?) for "bundle filelist" aka `BundleEntries`.  mildly distressed about about having three synonyms, now, for this entity.  meanwhile, "context" could either mean the golang type that essentially amounts to immutable maps or the datamon type that's currently a collection of GCS stores.

i don't have such an issue with these naming conventions when reading the source, yet i worry that conversationally these naming conventions will eventually have a negative impact on the user experience.

this isn't specific to this patch, so i'm not saying this comment need block merge.  feel free to resolve after reading.

Reply:
index stands for "file list file"